controllers/osbuildconfig_controller.go

```diff
@@ -70,15 +81,139 @@ func (r *OSBuildConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
-	err = r.OSBuildCRCreator.Create(ctx, osBuildConfig)
+	newOSBuildInstanceIsNeeded, err := r.checkIfNewOSBuildInstanceIsNeeded(ctx, osBuildConfig, logger)
+	if err != nil {
+		return ctrl.Result{Requeue: true, RequeueAfter: RequeueForShortDuration}, nil
+	}
+
+	if newOSBuildInstanceIsNeeded {
+		// create new OSBuild instance
+		targetBuild := osBuildConfig.Spec.Details.TargetImage.TargetImageType
+		if targetBuild == osbuilderv1alpha1.EdgeInstallerImageType {
+			// start with edge-container OSBuild instance
+			targetBuild = osbuilderv1alpha1.EdgeContainerImageType
+		}
+		return r.createOSBuildInstance(ctx, logger, osBuildConfig, targetBuild)
+	} else {
+		return r.updateOSBuildConfigCurrentStatus(ctx, logger, osBuildConfig)
+	}
+}
+
+func (r *OSBuildConfigReconciler) checkIfNewOSBuildInstanceIsNeeded(ctx context.Context, osBuildConfig *osbuilderv1alpha1.OSBuildConfig, logger logr.Logger) (bool, error) {
```

Comment:
Not sure how it can happen - it means that 2 different calls would be... 
And 2 new instances would be created

Reply:
It can happen just because of the way event handling is implemented. Imaging this scenario:
1. The node on which the operator is running fails and shuts down.
2. While OCP is rescheduling the operator the user updates the configuration, but also some automated procedure calls the webhook
3. The operator comes back to life examines the CR and as a result treats the config change disregarding the webhook timestamp change
4. At an unknown time in the future the operator restarts for some reason
5. The timestamp is in the status is still different than the one in the annotation and as a result another build is scheduled
This scenario is more elaborate and includes a downtime of the operator and hence might seem as an overreach. But, the downtime only comes to show that multiple things can change on the CR until the operator handles the event.

Now, I don't understand you comment about "2 instances would be created". Instances of what? If OSBuildConfig, then why? If OSBuild, then no, only one should be created

Reply:
I can understand it, It's really a rare condition 
I treated it anyway and added a UT